app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlController.kt

```diff
@@ -378,6 +380,27 @@ class DefaultSessionControlController(
             metrics.track(Event.PocketTopSiteClicked)
         }
 
+        val baseURL = url.tryGetHostFromUrl()
+        if (SupportUtils.GOOGLE_URL.contains(baseURL)) {
+            val availableEngines = activity
+                .components
+                .core
+                .store
+                .state
+                .search
+                .searchEngines
+
+            val searchAccessPoint = Event.PerformedSearch.SearchAccessPoint.TOPSITE
+            val event =
+                availableEngines.firstOrNull { engine -> engine.suggestUrl?.contains(baseURL) == true }
+                    ?.let { searchEngine ->
+                        searchAccessPoint.let { sap ->
+                            MetricsUtils.createSearchEvent(searchEngine, store, sap)
+                        }
+                    }
```

Comment:
At the end of the day this code searches the Google `SearchEngine` instance only, with the check above. Should we simplify this?

If we want to keep it generic for supporting more search engines in the future then:
* Should we check `resultUrls` instead of `suggestUrl` since the latter one is the URL we query search suggestions from and doesn't have to be related to the search engine URL. The result URLs may be closer.
* Should we also search in `*.state.search.availableEngines`? Otherwise you will not find a search engine if the user decided to hide it.
* In `appendSearchAttributionToUrlIfNeeded()` we already change the URL to the one with the search attribution, so we already know the context, it seems complicated to then check this here again. 🤔 This would also allow us to directly lookup the search engine (e.g. Google) and not have to search for it here, I assume?

Reply:
Very much agreed on this. I filed https://github.com/mozilla-mobile/fenix/issues/17956 for this work so that we can move this patch into production sooner. I sent out an email to explain my thoughts behind that.